metrics/collectd/collectd.bash

```diff
@@ -12,15 +12,21 @@ collectd_pod="collectd"
 init_stats() {
 	local wait_time=$1
 
-	# create collectd-config configmap
+	# create collectd-config configmap, delete old if there is one
+	kubectl get configmap collectd-config >/dev/null 2>&1 && kubectl delete configmap collectd-config
```

Comment:
heh heh - yeah, sure, coz, how many times have we been caught out by this ;-)
I wonder if you want to do the same for the collectd daemonset as well?

Reply:
Actually I'd prefer to leave the collectd daemonset creation as it is.

If the daemonset is already running and the configuration (collectd.yaml) has not changed, current code works fine:

```
# init_stats 10
configmap "collectd-config" deleted
configmap/collectd-config created
daemonset.apps/collectd unchanged
daemon set "collectd" successfully rolled out
```

On the other hand, if there's a change in yaml, for instance
```
# grep args -A1 *.yaml
          args:
            - echo my name is $(whoami) ; collectd -f;
```

current code nicely updates the existing daemonset
```
# init_stats 10
configmap "collectd-config" deleted
configmap/collectd-config created
daemonset.apps/collectd configured
Waiting for daemon set "collectd" rollout to finish: 1 out of 2 new pods have been updated...
Waiting for daemon set "collectd" rollout to finish: 1 out of 2 new pods have been updated...
Waiting for daemon set "collectd" rollout to finish: 1 of 2 updated pods are available...
daemon set "collectd" successfully rolled out
```

...and it indeed gets updated
```
# kubectl logs collectd-9qqsr
my name is root
plugin_load: plugin "aggregation" successfully loaded.
...
```

Reply:
ah, OK, understood - only the configmap clashes on re-creation, but the daemonset updates (or not) as necessary. understood.

Reply:
There's something still that I didn't really consider already. What happens if configmap has changed but yaml has not. That is, we'd like to change collectd configuration, but collectd daemonset happened to be already running... I'll test if we indeed should restart the daemonset anyway, just to make sure it's running the latest configuration.

Reply:
Checked it. Collectd daemonset needed a restart in order to apply new configuration. I added daemonset restart to the commit and force-pushed new version.

So it was very good point after all, thanks! :)